Having trouble connecting to postgres database when making migrations

After running docker-compose:
```
version: '3.1'
services:
  adminer:
    image: adminer
    restart: always
    ports:
      - 8080:8080
  database:
    image: "postgres"
    env_file:
      - database.env # configure postgres
```
```
// database.env
POSTGRES_USER=postgres
POSTGRES_PASSWORD=secret
POSTGRES_DB=syndicate_d
```
My terminal outputs:
```
...
database_1  | 2021-01-10 21:39:13.853 UTC [1] LOG:  database system is ready to accept connections
...
adminer_1   | [Sun Jan 10 21:39:12 2021] PHP 7.4.14 Development Server (http://[::]:8080) started
```

So far so good...

Next, I attempt to make my migrations using [golang-migrate](https://github.com/golang-migrate/migrate#cli-usage) CLI ❌ :
`migrate -source file://migrations -database postgres://postgres:secret@localhost/postgres?sslmode=disable up`

```
error: pq: role "postgres" does not exist
make: *** [migrate] Error 1
```

I am having a really hard time finding information on my circumstance. Help would be greatly appreciated!

--- a/.env
+++ b/.env
@@ -1,0 +1,3 @@
+POSTGRES_USER=user
+POSTGRES_PASSWORD=secret
+POSTGRES_DB=syndicate

--- a/Makefile
+++ b/Makefile
@@ -5,8 +5,8 @@ compose:
 
 migrate:
 	migrate -source file://migrations \
-			-database postgres://user:secret@localhost:5432/syndicate?sslmode=disable up
+			-database postgres://user:secret@db:5432/syndicate?sslmode=disable up
 
 migrate-down:
 	migrate -source file://migrations \
-			-database postgres://postgres:secret@localhost/postgres?sslmode=disable down
+			-database postgres://postgres:secret@db/postgres?sslmode=disable down

--- a/database.env
+++ b/database.env
@@ -1,3 +1,0 @@
-POSTGRES_PASSWORD=secret
-POSTGRES_DB=syndicate
-POSTGRES_USER=user

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -1,13 +1,28 @@
-version: '3.1'
+version: "3.8"
+
 services:
+  db:
+    image: postgres
+    environment:
+      POSTGRES_DB: $POSTGRES_DB
+      POSTGRES_USER: $POSTGRES_USER
+      POSTGRES_PASSWORD: $POSTGRES_PASSWORD
+
+  migrate:
+    image: migrate/migrate
+    volumes:
+      - ./migrations:/migrations
+    depends_on:
+      - db
+    command: -source=file://migrations -database postgres://$POSTGRES_USER:$POSTGRES_PASSWORD@db:5432/$POSTGRES_DB?sslmode=disable up
+    restart: on-failure
+
   adminer:
     image: adminer
     restart: always
     ports:
-      - 8080:8080
-  db:
-    image: "postgres"
-    env_file:
-      - database.env # configure postgres
-    ports:
-      - 5432:5432
+      - "8080:8080"
+    environment:
+      ADMINER_DEFAULT_SERVER: db
+    depends_on:
+      - db

